io/common/bootlist_test.py

```diff
@@ -18,38 +20,46 @@
 from avocado import main
 from avocado import Test
 from avocado.utils import process
+from avocado import skipUnless
+
+IS_POWER_VM = 'pSeries' in open('/proc/cpuinfo', 'r').read()
 
 
 class BootlisTest(Test):
     '''
     Displays and alters the list of boot devices available
     to the system
     '''
-
+    @skipUnless(IS_POWER_VM,
+                "supported only on PowerVM platform")
     def setUp(self):
         '''
         To check and interfaces
         '''
         self.host_interfaces = self.params.get("host_interfaces",
-                                               default="").split(",")
-        if not self.host_interfaces:
-            self.cancel("user should specify host interfaces")
-        self.interfaces = ""
-        interfaces = netifaces.interfaces()
-        for host_interface in self.host_interfaces:
-            if host_interface not in interfaces:
-                self.cancel("interface is not available")
-            self.interfaces = '%s %s' % (self.interfaces, host_interface)
-        cmd = "cat /proc/cpuinfo | grep \'pSeries\'"
-        if process.system(cmd, shell=True, verbose=True,
-                          ignore_status=True):
-            self.cancel("Test not supported on the PowerNV Host platform")
+                                               default=None)
+        self.disk_names = self.params.get("disks", default=None)
+        if self.host_interfaces is not None:
+            if not self.host_interfaces:
+                self.cancel("user should specify host interfaces")
+            self.interfaces = self.host_interfaces.replace(',', ' ')
+            interfaces = netifaces.interfaces()
+            for host_interface in self.host_interfaces.split(","):
+                if host_interface not in interfaces:
+                    self.cancel("interface is not available")
+        if self.disk_names is not None:
+            if not self.disk_names:
+                self.cancel("user should specify disk name")
+            self.disks = self.disk_names.replace(',', ' ')
 
     def bootlist_mode(self, param):
         '''
         converting into different mode
         '''
-        cmd = "bootlist -m %s %s" % (param, self.interfaces)
+        if self.host_interfaces is not None:
```

Comment:
is it necessary to check `self.host_interfaces` here, as you are already checking in the `setUp`

Reply:
yes it necessary,because based on that we are running the command for all bootlist mode for both and passing self.interfaces and self.disks based on that.